### DISCUSSION TITLE
SSH Tunnel for database

### DISCUSSION THREAD

**Discussion by Lan-Phan** [illacloud/illa-builder#717] [General]
Hello,

As ILLA uses GORM to connect to db, I think we can easily add SSH Tunnel support for db access, what do you think?

Here is an example about SSH Tunnel configuration for db access: https://docs.uibakery.io/data-sources/ssh-tunneling

**Comment by naj1n** [comment]
It's an ambiguous statement about which database to link to. Please provide more information about it.

**Reply by Lan-Phan** [reply]
Can we offer SSH Tunnel for Postgresql db first?

In fact, SSH Tunnel is applicable for all databases

**Reply by naj1n** [reply]
Is this requirement for the resource integration in illa builder?

**Reply by Lan-Phan** [reply]
yes @naj1n ,

Please take a look at attached image, this is a configuration for Postgres db access. In Advanced Options, there is only 1 option about SSL, I think we can add another option "SSH Tunnel" there. I also attached another image about Jet Admin for your reference, the way they allow to configure SSH Tunnel for Postgres

<img width="1183" alt="Screen Shot 2022-12-28 at 12 34 40 PM" src="https://user-images.githubusercontent.com/106818620/209763007-84afd644-6561-4cf0-8e86-ae466566e2be.png">

<img width="1067" alt="Screen Shot 2022-12-28 at 12 38 44 PM" src="https://user-images.githubusercontent.com/106818620/209763427-41a0670d-e08e-4fe0-abf7-6ae897df74e5.png">

**Reply by naj1n** [reply]
@Lan-Phan Thanks for the feature request! We'll get this prioritized and ship this out soon.

**Comment by Lan-Phan** [comment]
Hi @naj1n 

Thanks. Do you have any estimation about time this feature will be shipped?
This morning, I checked your live demo site and got issue, already reported it here: https://github.com/illacloud/illa-builder/issues/756

**Reply by naj1n** [reply]
Sorry, I don't have a clear time expectation for this FR. Of course, we'll fix the live demo bug you mentioned ASAP.
